src/Client/PullPayment.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BTCPayServer\Client;
+
+class PullPayment extends AbstractClient
+{
+    public function getStorePullPayments(
+        string $storeId,
+        bool $includeArchived
+    ): \BTCPayServer\Result\PullPaymentList {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $body = json_encode(
+            [
+                'includeArchived' => $includeArchived,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\PullPaymentList(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function createPullPayment(
+        string $storeId,
+        string $paymentName,
+        string $paymentAmount,
+        string $paymentCurrency,
+        mixed $paymentPeriod,
+        mixed $boltExpiration,
+        mixed $startsAt,
+        mixed $expiresAt,
+        array $paymentMethods
+    ): \BTCPayServer\Result\PullPayment {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'name' => $paymentName,
+                'amount' => $paymentAmount,
```

Comment:
because we use PreciseNumber we need to do `__toString()` here
`'amount' => $paymentAmount->__toString(),`